targets/targets.json

```diff
@@ -8730,17 +8729,48 @@
             "FLASH",
             "MPU"
         ],
-        "components_add": ["FLASHIAP"],
         "detect_code": ["1305"],
         "release_versions": ["5"],
         "device_name": "M2351KIAAEES",
         "bootloader_supported": true,
         "tickless-from-us-ticker": true,
         "forced_reset_timeout": 3,
-        "mbed_rom_start"    : "0x10040000",
-        "mbed_rom_size"     : "0x40000",
-        "mbed_ram_start"    : "0x30008000",
-        "mbed_ram_size"     : "0x10000"
+        "public": false
+    },
+    "NU_PFM_M2351_NS": {
+        "inherits": ["NU_PFM_M2351_CM"],
+        "core": "Cortex-M23-NS",
+        "trustzone": true,
+        "extra_labels_add": [
+            "M23_NS",
+            "NU_PREBUILD_SECURE"
+        ],
+        "macros_add": [
+            "MBED_TZ_DEFAULT_ACCESS=1"
+        ],
+        "components_add": ["FLASHIAP"],
+        "post_binary_hook": {"function": "M2351Code.merge_secure"},
+        "secure_image_filename": "NuMaker-mbed-TZ-secure-example.hex",
+        "mbed_rom_start"        : "0x10040000",
+        "mbed_rom_size"         : "0x40000",
+        "mbed_ram_start"        : "0x30008000",
+        "mbed_ram_size"         : "0x10000"
+    },
+    "NU_PFM_M2351_S": {
+        "inherits": ["NU_PFM_M2351_CM"],
+        "core": "Cortex-M23",
+        "trustzone": true,
+        "extra_labels_add": [
+            "M23_S"
+        ],
+        "device_has_remove": ["TRNG"],
+        "components_add": ["FLASHIAP"],
+        "deliver_to_target": "NU_PFM_M2351_NS",
+        "delivery_dir": "TARGET_NUVOTON/TARGET_M2351/TARGET_M23_NS/TARGET_NU_PFM_M2351_NS/TARGET_NU_PREBUILD_SECURE",
```

Comment:
@ccli8 Currently in mbed-os, PSA targets are named as targetname_S and targetname_NS for secure and non-secure respectively. Unfortunately, the proposed names are not following that. Due to that, I suggest the following,

```
NU_PFM_M2351_S: Target name for building M2351 TFM secure code
NU_PFM_M2351_NS: Target name for building M2351 TFM non-secure code
NU_PFM_M2351_NPSA_S: Target name for building M2351 non-TFM secure code
NU_PFM_M2351_NPSA_NS: Target name for building M2351 non-TFM non-secure code
```

Can you please update the target names and the respective folders?

Reply:
@devran01 It concerns something. First, #11467 doesn't support **NPSA** suffix. Second, **NU_PFM_M2351_NPSA_NS** is exactly 20 chars. Not sure if it breaks online database max 20 chars limit. If possible, prefer keeping the current naming.

Reply:
@ccli8 Thanks for highlighting potential issue with new naming and #11467. We need to fix that as well. 
But before that, @mark-edgeworth Is it ok to have target names with exactly 20 characters or should they be less than 20?

Unfortunately we cannot retain the current names as this will cause confusion with other existing PSA/TFM targets (https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L5678, https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L5700, https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L2115, https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L2168)

Reply:
@devran01 OK. I will rename as suggested.

Reply:
There is at least one other name in the database that has 20 characters, so this should be ok.